presto-hive/src/main/java/com/facebook/presto/hive/rule/HiveParquetDereferencePushDown.java

```diff
@@ -329,18 +337,22 @@ public PlanNode visitProject(ProjectNode project, Void context)
             if (!isParquetDereferenceEnabled(session, tableScan.getTable())) {
                 return visitPlan(project, context);
             }
+            Map<String, HiveColumnHandle> regularHiveColumnHandles = new HashMap<>();
+            regularHiveColumnHandles.putAll(tableScan.getAssignments().entrySet().stream()
+                    .collect(toMap(e -> e.getKey().getName(), e -> (HiveColumnHandle) e.getValue())));
+            regularHiveColumnHandles.putAll(tableScan.getAssignments().values().stream()
```

Comment:
Do we need this mapping (columnName in HiveColumnHandle -> HiveColumnHandle)? Expressions should be referring to the name in `Assignments`. Is that not the case in some cases?

Reply:
Sorry are you referring to the second `putAll` of the two here? If remove this mapping, I would get 
`nested column [X.Y]'s base column X is not present in table scan output` 
error in the immediate [following logic](https://github.com/prestodb/presto/blob/release-0.247/presto-hive/src/main/java/com/facebook/presto/hive/rule/HiveParquetDereferencePushDown.java#L356). This places uses the subfield name to look up back to `HiveColumnHandle`, and now subfile has the same name as the original `HiveColumnHandle`, so I used the same map to do this lookup.